src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -106,6 +107,33 @@ @implementation GRPCCall {
 // TODO(jcanizales): If grpc_init is idempotent, this should be changed from load to initialize.
 + (void)load {
   grpc_init();
+  callFlags = [NSMutableDictionary dictionary];
+}
+
++ (void)setCallAttribute:(GRPCCallAttr)callAttr host:(NSString *)host path:(NSString *)path {
+  NSString *hostAndPath = [NSString stringWithFormat:@"%@%@", host, path];
+  switch (callAttr) {
+    case GRPCCallAttrDefault:
+      callFlags[hostAndPath] = @0;
+      break;
+    case GRPCCallAttrIdempotentRequest:
+      callFlags[hostAndPath] = @GRPC_INITIAL_METADATA_IDEMPOTENT_REQUEST;
+      break;
+    case GRPCCallAttrCacheableRequest:
+      callFlags[hostAndPath] = @GRPC_INITIAL_METADATA_CACHEABLE_REQUEST;
+      break;
+    default:
+      break;
+  }
+}
+
++ (uint32_t)getCallFlag:(NSString *)host path:(NSString *)path {
```

Comment:
`callFlagsForHost: path:`. With `getCallFlag:`, the parameter would be understood to be a flag. And methods should only start with "get" [if they return values indirectly via pointers passed as arguments](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html).